ProjectX/Foundation/Networking/Sources/Networking/Endpoint.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  NetworkEndpoint.swift
+//  
+//
+//  Created by Felipe Dias Pereira on 30/07/20.
+//
+import Foundation
+
+// MARK: Endpoint
+
+public enum BodyParameter {
+    case data(Data)
+    case dictionary([String: Any], options: JSONSerialization.WritingOptions = [])
+    case encodable(Encodable, encoder: JSONEncoder = .init())
+}
+
+public struct Endpoint<Kind: EndpointKind> {
+    public let path: String
+    public var method: Method = .get
+    public let urlQueries: [String: String]?
+    public let headers: [String: String]?
+    public let bodyParameter: BodyParameter?
+
+    public enum Method: String {
+        case post = "POST"
+        case get = "GET"
+        case put = "PUT"
+        case patch = "PATCH"
+        case delete = "DELETE"
+    }
+
+    public init(
+        path: String,
+        method: Endpoint<Kind>.Method = .get,
+        urlQueries: [String: String]? = nil,
+        headers: [String: String]? = nil,
+        bodyParameter: BodyParameter? = nil
+    ) {
+        self.path = path
+        self.method = method
+        self.urlQueries = urlQueries
+        self.headers = headers
+        self.bodyParameter = bodyParameter
+    }
+}
+
+// MARK: - Erros
+extension Endpoint {
+    enum Errors: LocalizedError {
+        case URLMalformated
+    }
+}
+
+extension Endpoint {
+    func makeRequest(host: String, with data: Kind.RequestData) throws -> URLRequest {
+        var components = URLComponents()
```

Comment:
### Suggestion
* To be able to test this method, what do you think of injecting this `URLComponents()` as a dependency injection through `init`?